### PR TITLE
Move dataKey to cmd parameters

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description:**
- Make dataKey a cmd parameter

### CODE REVIEW

File: controllers/rule_controller.go [hunk 1]
```diff
@@ -46,12 +45,14 @@ type RuleReconciler struct {
 	Log              logr.Logger
 	RuleConfigmap    types.NamespacedName
 	ValidationConfig validation.Config
+	RulesFileName    string
```
Comment (Tomasz-Smelcerz-SAP) [inline controllers/rule_controller.go]: I wouldn't use "RulesFileName" here. Controller actually doesn't care if this map is mounted anywhere - it's domain is to create a valid ConfigMap. `dataKey` or `mapKey` should be sufficient.

File: main.go [hunk 2]
```diff
@@ -59,12 +60,14 @@ func main() {
...
 	var rulesConfigmapName string
 	var rulesConfigmapNamespace string
+	var rulesFileName string
 
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.StringVar(&rulesConfigmapName, "rulesConfigmapName", "oathkeeper-rules", "Name of the Configmap that stores Oathkeeper rules.")
 	flag.StringVar(&rulesConfigmapNamespace, "rulesConfigmapNamespace", "oathkeeper-k8s-controller-system", "Namespace of the Configmap that stores Oathkeeper rules.")
+	flag.StringVar(&rulesFileName, "rulesFileName", "access_rules.json", "Name of the file with converted Oathkeeper rules")
```
Comment (Tomasz-Smelcerz-SAP) [inline main.go]: The controller doesn't create any file (yet), it's just a name of the key in the map. I think that config flag should be named accordingly: `mapKey` `keyName` or something like that.

File: main.go [hunk 3]
```diff
@@ -87,6 +90,7 @@ func main() {
 		Log:              ctrl.Log.WithName("controllers").WithName("Rule"),
 		RuleConfigmap:    types.NamespacedName{Name: rulesConfigmapName, Namespace: rulesConfigmapNamespace},
 		ValidationConfig: validationConfig,
+		RulesFileName:    rulesFileName,
```
Comment (Tomasz-Smelcerz-SAP) [inline main.go]: I think there should be a basic validation here. I think K8s uses this regex for key names: `[-._a-zA-Z0-9]+` Controller should fail on startup (os.Exit(1)) if the parameter does not match the validation regexp

**Comment by aeneasr** [comment]
Stellar Job! :)(

**Pull request merged**